### PR TITLE
Use CLI to assign team roles

### PR DESCRIPTION
**Pull request by lzdanski**
Resolves https://github.com/astronomer/docs/issues/2750

### CODE REVIEW

File: astro/manage-teams.md [hunk 1]
```diff
@@ -51,6 +51,57 @@ You can now [add your Team to a Workspace](manage-teams.md#add-a-team-to-a-works
...
+
+    This command returns the Team ID, team name, creation date, and Organization Role the Team has.
+
+4. Use the Team ID to assign the Team to other Workspaces in your Organization and optionally configure their [Workspace Role](/astro/user-permissions.md#workspace-roles).
+
+    ```sh
+    astro workspace team add <team-id> --role
+    ```
+
+    Available Role options include `WORKSPACE_MEMBER`, `WORKSPACE_OPERATOR`, or `WORKSPACE_OWNER`. By default, the Team is assigned to `WORKSPACE_MEMBER`.
```
Comment (jwitz) [inline astro/manage-teams.md:97]:
```suggestion
    Available role options include `WORKSPACE_MEMBER`, `WORKSPACE_AUTHOR`, `WORKSPACE_OPERATOR`, or `WORKSPACE_OWNER`. By default, the Team is assigned as a `WORKSPACE_MEMBER`.
```

File: astro/manage-teams.md [hunk 2]
```diff
@@ -51,6 +51,57 @@ You can now [add your Team to a Workspace](manage-teams.md#add-a-team-to-a-works
 
 4. Select the **Team** you want to add and define their **Workspace Role**, which determines their [Workspace user permissions](/astro/user-permissions.md#workspace-roles).
 
+## Add Teams to Workspaces with Astro CLI
```
Comment (jwitz) [inline astro/manage-teams.md:54]:
```suggestion
## Add Teams to Workspaces using the Astro CLI
```
We can keep this topic for now, but in the future I would expect something like this in the Automation/ CI/CD menu in an "Examples" doc with a proper scripting example for adding multiple Teams at once. Currently the topic is positioned as if the steps are what you'd do to programmatically add teams, but the actual automation process is more complicated and uses alternative authentication steps.

File: astro/manage-teams.md [hunk 3]
```diff
@@ -51,6 +51,57 @@ You can now [add your Team to a Workspace](manage-teams.md#add-a-team-to-a-works
...
+
```
Comment (jwitz) [inline astro/manage-teams.md:55]:
```suggestion
:::cli

You can add a Team to multiple Workspaces programmatically using the Astro CLI. See [`astro workspace team add`](cli/astro-workspace-team-add.md) for example output and commands.

:::

```
If you agree with my point above and don't want to publish this topic quite yet, we can use this note in its place under "Add a Team to a Workspace". I'm cool either way!

File: astro/manage-teams.md [hunk 4]
```diff
@@ -51,6 +51,57 @@ You can now [add your Team to a Workspace](manage-teams.md#add-a-team-to-a-works
...
+You can add a Team to Workspaces programmatically with the Astro CLI instead of individually adding them with the Cloud UI. You must be logged into the correct Organization to add a Team its Workspaces.
+
+### Prerequisites
+
+- Install the [Astro CLI](cli/install-cli.md).
+- Create a [Team](#create-a-team)
```
Comment (jwitz) [inline astro/manage-teams.md:61]:
```suggestion
- The [Astro CLI](cli/install-cli.md).
- A [Team](#create-a-team)
```

File: astro/manage-teams.md [hunk 5]
```diff
@@ -51,6 +51,57 @@ You can now [add your Team to a Workspace](manage-teams.md#add-a-team-to-a-works
...
+1. Log in to Astro. 
+
+    ```sh
+    astro login astronomer.io
+    ```
+
+2. Choose the Workspace where you want to add a Team.
+    
+    ```sh
+    astro workspace
```
Comment (jwitz) [inline astro/manage-teams.md:74]:
```suggestion
    astro workspace list
```

File: astro/manage-teams.md [hunk 6]
```diff
@@ -51,6 +51,57 @@ You can now [add your Team to a Workspace](manage-teams.md#add-a-team-to-a-works
...
+    ```
+
+    :::tip
```
Comment (jwitz) [inline astro/manage-teams.md:77]:
```suggestion
```
Tip is more "cool and optional", this feels like more of a troubleshooting step

File: astro/manage-teams.md [hunk 7]
```diff
@@ -51,6 +51,57 @@ You can now [add your Team to a Workspace](manage-teams.md#add-a-team-to-a-works
...
+
+    If you do not see the Workspace as a choice, confirm which Organization you're currently working in with `astro organization list`. You can switch to the correct Organization with `astro organization switch`. You can also change your current Organization in the Cloud UI.
```
Comment (jwitz) [inline astro/manage-teams.md:79]:
```suggestion
    If you don't see the right Workspace as a choice, confirm which Organization you're currently working in using `astro organization list`. You can switch to the correct Organization using `astro organization switch`.
```

File: astro/manage-teams.md [hunk 8]
```diff
@@ -51,6 +51,57 @@ You can now [add your Team to a Workspace](manage-teams.md#add-a-team-to-a-works
...
+
+    :::
+
+3. Retrieve the Team ID.
```
Comment (jwitz) [inline astro/manage-teams.md:83]:
```suggestion
3. Run the following command to find the Team you want to add. Copy its `TEAM ID`:
```

File: astro/manage-teams.md [hunk 9]
```diff
@@ -51,6 +51,57 @@ You can now [add your Team to a Workspace](manage-teams.md#add-a-team-to-a-works
...
+You can add a Team to Workspaces programmatically with the Astro CLI instead of individually adding them with the Cloud UI. You must be logged into the correct Organization to add a Team its Workspaces.
+
+### Prerequisites
+
+- Install the [Astro CLI](cli/install-cli.md).
+- Create a [Team](#create-a-team)
+
+### Add Teams to Workspaces
+
+1. Log in to Astro.
```
Comment (jwitz) [inline astro/manage-teams.md:65]:
```suggestion
1. Run the following command to log in to Astro:
```

File: astro/manage-teams.md [hunk 10]
```diff
@@ -51,6 +51,57 @@ You can now [add your Team to a Workspace](manage-teams.md#add-a-team-to-a-works
...
+
+### Add Teams to Workspaces
+
+1. Log in to Astro. 
+
+    ```sh
+    astro login astronomer.io
+    ```
+
+2. Choose the Workspace where you want to add a Team.
```
Comment (jwitz) [inline astro/manage-teams.md:71]:
```suggestion
2. Run the following command to identify the Workspace where you want to add the Team.
```
This is one of the steps where it wouldn't really make sense to have it from a programmatic standpoint, they would likely have the Workspace ID on hand or need to parse this output to retrieve it.

File: astro/manage-teams.md [hunk 11]
```diff
@@ -51,6 +51,57 @@ You can now [add your Team to a Workspace](manage-teams.md#add-a-team-to-a-works
...
+
+    :::
+
+3. Retrieve the Team ID. 
+
+    ```sh
+    astro organization team list
+    ```
+
+    This command returns the Team ID, team name, creation date, and Organization Role the Team has.
```
Comment (jwitz) [inline astro/manage-teams.md:89]:
```suggestion
```

File: astro/manage-teams.md [hunk 12]
```diff
@@ -51,6 +51,57 @@ You can now [add your Team to a Workspace](manage-teams.md#add-a-team-to-a-works
...
+
+4. Use the Team ID to assign the Team to other Workspaces in your Organization and optionally configure their [Workspace Role](/astro/user-permissions.md#workspace-roles).
```
Comment (jwitz) [inline astro/manage-teams.md:91]:
```suggestion
4. Run the following command to assign the Team to other Workspaces in your Organization.
```

File: astro/manage-teams.md [hunk 13]
```diff
@@ -51,6 +51,57 @@ You can now [add your Team to a Workspace](manage-teams.md#add-a-team-to-a-works
...
+
+    ```sh
+    astro workspace team add <team-id> --role
+    ```
+
+    Available Role options include `WORKSPACE_MEMBER`, `WORKSPACE_OPERATOR`, or `WORKSPACE_OWNER`. By default, the Team is assigned to `WORKSPACE_MEMBER`.
+
+5. You can test that you assigned your Team successfully by listing all Teams in your current Workspace.
```
Comment (jwitz) [inline astro/manage-teams.md:99]:
```suggestion
5. Run the following command to confirm that the Team was successfully added to your current Workspace.
```

File: astro/user-permissions.md [hunk 14]
```diff
@@ -80,10 +80,11 @@ To manage a user's Workspace permissions, see [Manage Worksapce users](manage-wo
 
 ## User roles and Team roles
```
Comment (jwitz) [inline astro/user-permissions.md:81]:
```suggestion
## Relationship between user roles and Team roles
```
Hmm, I'm not sure the edits here relate to the topic. When I wrote this, it was meant to clarify the logic of user roles overriding Team roles and vice versa. Tried updating the title to make this clear.

File: astro/user-permissions.md [hunk 15]
```diff
@@ -80,10 +80,11 @@ To manage a user's Workspace permissions, see [Manage Worksapce users](manage-wo
...
 
-There are two ways to define a user's role in a Workspace:
+You can assign Teams to Workspaces and configure their Workspace user roles in either the Cloud UI or programmatically with the Astro CLI:
 
 - Define the individual user role when you [add a user](manage-workspace-users.md#add-a-user-to-a-workspace) to a Workspace.
 - Assign a Workspace role to a [Team](manage-teams.md#add-a-team-to-a-workspace).
```
Comment (jwitz) [inline astro/user-permissions.md:86]:
```suggestion
- Assign a Workspace role to a [Team](manage-teams.md#add-a-team-to-a-workspace) and add the user to the Team.
```

File: astro/user-permissions.md [hunk 16]
```diff
@@ -80,10 +80,11 @@ To manage a user's Workspace permissions, see [Manage Worksapce users](manage-wo
...
+- Assign Teams to Workspaces and define their user roles with the [Astro CLI](manage-teams.md#add-teams-to-workspaces-with-astro-cli).
```
Comment (jwitz) [inline astro/user-permissions.md:87]:
```suggestion
```

File: astro/user-permissions.md [hunk 17]
```diff
@@ -80,10 +80,11 @@ To manage a user's Workspace permissions, see [Manage Worksapce users](manage-wo
 
 ## User roles and Team roles
 
-There are two ways to define a user's role in a Workspace:
+You can assign Teams to Workspaces and configure their Workspace user roles in either the Cloud UI or programmatically with the Astro CLI:
```
Comment (jwitz) [inline astro/user-permissions.md:83]:
```suggestion
There are two ways to define a user's role in a Workspace:
```

**Review by jwitz — CHANGES_REQUESTED** [review]
Reference updates look good! Requesting changes on some thoughts I had in the new process topic, curious to know what you think before moving ahead

File: astro/manage-teams.md (hunk 3 above)
Comment (lzdanski) [reply astro/manage-teams.md:55]: Would it follow an analogous process to the Adding multiple users to a workspace? https://docs.astronomer.io/astro/manage-workspace-users#add-a-group-of-users-to-a-workspace-using-the-astro-cli

File: astro/manage-teams.md [hunk 18]
```diff
@@ -51,6 +51,53 @@ You can now [add your Team to a Workspace](manage-teams.md#add-a-team-to-a-works
...
+
+2. Run the following command to identify the Workspace where you want to add the Team.
+    
+    ```sh
+    astro workspace list
+    ```
+
+    If you don't see the right Workspace as a choice, confirm which Organization you're currently working in using `astro organization list`. You can switch to the correct Organization using `astro organization switch`.
+
+    :::
```
Comment (lzdanski) [inline astro/manage-teams.md:79]:
```suggestion
```

File: astro/manage-teams.md (hunk 3 above)
Comment (lzdanski) [reply astro/manage-teams.md:55]: Have discussed with CLI team that these processes are _not_ analogous, and this functionality would be really tricky to script. Will  skip for now.

File: astro/manage-teams.md [hunk 19]
```diff
@@ -51,6 +51,10 @@ You can now [add your Team to a Workspace](manage-teams.md#add-a-team-to-a-works
 
 4. Select the **Team** you want to add and define their **Workspace Role**, which determines their [Workspace user permissions](/astro/user-permissions.md#workspace-roles).
 
+:::cli
+You can add a Team to multiple Workspaces programmatically using the Astro CLI. See [`astro workspace team add`](cli/astro-workspace-team-add.md) for example output and commands.
+:::
```
Comment (jwitz) [inline astro/manage-teams.md:56]:
```suggestion
:::cli

You can add a Team to multiple Workspaces programmatically using the Astro CLI. See [`astro workspace team add`](cli/astro-workspace-team-add.md) for example output and commands.

:::
```

File: astro/cli/astro-workspace-team.md [hunk 20]
```diff
@@ -38,9 +38,8 @@ You can retrieve a Team's ID in one of two ways:
 
 | Option                        | Description                       | Possible Values                                                                   |
 | ----------------------------- | --------------------------------- | --------------------------------------------------------------------------------- |
```
Comment (jwitz) [inline astro/cli/astro-workspace-team.md:40]: Software commands behave a bit differently from Astro commands, it's possible that `--workspace-id` is required but I can't verify at the moment. I would err on the side of caution and keep this as-is for now (unless you tested on Software, in which case go ahead with this!)

**Review by jwitz — APPROVED** [review]
LGTM! Also if you are using VScode as your code editor, I recommend installing [prettier](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode) and running alt + shift + f to automatically space your tables for future raw markdown legibility

**Pull request merged**